compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -1122,14 +1122,14 @@ impl<'hir> LoweringContext<'_, 'hir> {
 
         let mut used_input_regs = FxHashMap::default();
         let mut used_output_regs = FxHashMap::default();
+        let mut required_features: Vec<&str> = vec![];
         for (idx, op) in operands.iter().enumerate() {
             let op_sp = asm.operands[idx].1;
```

Comment:
can you instead of clearing at the end eagerly clear here?
```suggestion
            required_features.clear();
            let op_sp = asm.operands[idx].1;
```

Reply:
Sure, but what difference would it make? Are you thinking about anything particular besides readability?

Reply:
safety, if we clear at the end it's easier to accidentally forget one branch and reuse a filled vec.

rn I always have to look at the whole loop to see that this is correct while when clearing at the start is "obviously" correct